models/migrations/v126.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/builder"
+	"xorm.io/xorm"
+)
+
+func fixTopicRepositoryCount(x *xorm.Engine) error {
+	type Topic struct {
+		ID        int64
+		RepoCount int
+	}
+
+	_, err := x.Exec(builder.Delete(builder.NotIn("`repo_id`", builder.Select("`id`").From("`repository`"))).From("`repo_topic`"))
+	if err != nil {
+		return err
+	}
+
+	var last int
+	const batchSize = 50
+	for {
```

Comment:
How about directly update set sub query.

Reply:
subquery requires ID from updateble table, I don't know if xorm supports such updates

Reply:
How about?
```
_, err := x.Exec("UPDATE topic SET repo_count = (SELECT COUNT(*) FROM repo_topic WHERE repo_topic.topic_id = topic.id)");
```

Worked for me in PSQL, SQLite, MySQL and MSSQL.

Reply:
AFAIU xorm does not support `INSERT` or `UPDATE` without going through specific records, as expected from an ORM. Only bulk DML operation supported is `DELETE`.

On second thought, it should be interesting to check whether the following is supported, though:

```
_, err := x.Cols("repo_count").
	SetExpr("repo_count", builder.Select("count(*)").From("repo_topic").Where(
		builder.Eq{"repo_topic.topic_id": "topic.ID"},
	)).Update(new(Topic))
```

Reply:
I think it would treat `topic.ID` as string value